shell/platform/android/io/flutter/embedding/engine/mutatorsstack/FlutterMutatorView.java

```diff
@@ -0,0 +1,89 @@
+package io.flutter.embedding.engine.mutatorsstack;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Matrix;
+import android.graphics.Path;
+import android.widget.FrameLayout;
+import androidx.annotation.NonNull;
+
+public class FlutterMutatorView extends FrameLayout {
+
+  private FlutterMutatorsStack mutatorsStack;
+  private float screenDensity;
+  private int left;
+  private int top;
+
+  public FlutterMutatorView(@NonNull Context context) {
+    super(context, null);
+  }
+
+  public void readyToDisplay(
+      @NonNull FlutterMutatorsStack mutatorsStack,
+      float screenDensity,
+      int left,
+      int top,
+      int width,
+      int height) {
+    this.mutatorsStack = mutatorsStack;
+    this.screenDensity = screenDensity;
+    this.left = left;
+    this.top = top;
+    FrameLayout.LayoutParams layoutParams = new FrameLayout.LayoutParams(width, height);
+    layoutParams.leftMargin = left;
+    layoutParams.topMargin = top;
+    setLayoutParams(layoutParams);
+    setClipChildren(false);
+    setWillNotDraw(false);
+  }
+
+  @Override
+  public void draw(Canvas canvas) {
+    // Apply all clippings on the parent canvas.
+    canvas.save();
+    for (Path path : mutatorsStack.getFinalClippingPaths()) {
+      // Reverse the current offset.
+      //
+      // The frame of this view includes the final offset of the bounding rect.
+      // We need to apply all the mutators to the view, which includes the mutation that leads to
+      // the final offset. We should reverse this final offset, both as a translate mutation and to
+      // all the clipping paths
+      Path pathCopy = new Path(path);
+      pathCopy.offset(-left, -top);
+      canvas.clipPath(pathCopy);
+    }
+    super.draw(canvas);
+    canvas.restore();
+  }
+
+  @Override
+  public void dispatchDraw(Canvas canvas) {
+    // Apply all the transforms on the child canvas.
+    canvas.save();
```

Comment:
do you think that there might be other transforms that we can want to preserve?

Reply:
what do you mean?